src/code/z_play_hireso.c

```diff
@@ -1,27 +1,1349 @@
+            }
+            color += 3;
+        }
+    }
+
+    gDPPipeSync(gfx++);
+    *gfxP = gfx;
+}
+
+TexturePtr sBombersNotebookEventIconTextures[] = {
+    gBombersNotebookEntryIconExclamationPointTex,
+    &D_0700AC00,
+    &D_0700AEA0,
+};
+s16 sBombersNotebookEntryIconColors[][3] = {
+    { 255, 255, 0 },
+    { 141, 255, 182 },
+    { 255, 100, 60 },
+};
+
+void BombersNotebook_DrawEntries(Gfx** gfxP, s32 row, u32 rectTop) {
+    Gfx* gfx = *gfxP;
+    s32 entryRectLeft;
+    u32 eventIcon;
+    u32 unfinishedEvent = false;
+    u16 startTime;
+    u16 endTime;
+    s32 j = 0;
+    u32 iconRectLeft;
+    s32 entryRectRight;
+    u32 yOffset;
+    s32 entryWidth;
+
+    while (true) {
+        if (sBombersNotebookEntries[row][j] == BOMBERS_NOTEBOOK_ENTRY_END) {
+            if (!unfinishedEvent) {
+                gDPPipeSync(gfx++);
+                gDPSetCombineLERP(gfx++, PRIMITIVE, ENVIRONMENT, TEXEL0, ENVIRONMENT, TEXEL0, 0, PRIMITIVE, 0,
+                                  PRIMITIVE, ENVIRONMENT, TEXEL0, ENVIRONMENT, TEXEL0, 0, PRIMITIVE, 0);
+                gDPSetPrimColor(gfx++, 0, 0, 0, 0, 0, 255);
+                gDPSetEnvColor(gfx++, 0, 0, 0, 255);
+                gDPLoadTextureBlock(gfx++, gBombersNotebookStampTex, G_IM_FMT_IA, G_IM_SIZ_8b, 32, 48, 0,
+                                    G_TX_MIRROR | G_TX_WRAP, G_TX_MIRROR | G_TX_WRAP, G_TX_NOMASK, G_TX_NOMASK,
+                                    G_TX_NOLOD, G_TX_NOLOD);
+                BombersNotebook_TextureRectangle(&gfx, 567 * 4, (rectTop + 1) * 4, 599 * 4, (rectTop + 49) * 4, 0, 0, 0,
+                                                 0x400, 0x400);
+                gDPPipeSync(gfx++);
+                gDPSetPrimColor(gfx++, 0, 0, 255, 255, 255, 255);
+                gDPSetEnvColor(gfx++, 255, 0, 0, 255);
+                BombersNotebook_TextureRectangle(&gfx, 565 * 4, (rectTop + -1) * 4, 597 * 4, (rectTop + 47) * 4, 0, 0,
+                                                 0, 0x400, 0x400);
+                gDPPipeSync(gfx++);
+                gDPSetCombineMode(gfx++, G_CC_MODULATEIA_PRIM, G_CC_MODULATEIA_PRIM);
+                gDPSetEnvColor(gfx++, 0, 0, 0, 255);
+            }
+            break;
+        }
+
+        if (sBombersNotebookEntries[row][j] & BOMBERS_NOTEBOOK_ENTRY_POS_ABOVE) {
+            yOffset = 8;
+        } else if (sBombersNotebookEntries[row][j] & BOMBERS_NOTEBOOK_ENTRY_POS_BELOW) {
+            yOffset = 24;
+        } else { // BOMBERS_NOTEBOOK_ENTRY_POS_CENTER
+            yOffset = 16;
+        }
+
+        startTime = BOMBERS_NOTEBOOK_ENTRY_GET_START_TIME(row, j) - CLOCK_TIME(6, 0);
+        endTime = BOMBERS_NOTEBOOK_ENTRY_GET_END_TIME(row, j) - CLOCK_TIME(6, 0);
+        entryRectLeft = sBombersNotebookDayRectRectLeft[BOMBERS_NOTEBOOK_ENTRY_GET_DAY(row, j)] + (startTime / 455);
+        entryRectRight = sBombersNotebookDayRectRectLeft[BOMBERS_NOTEBOOK_ENTRY_GET_DAY(row, j)] + (endTime / 455);
```

Comment:
```suggestion
        entryRectLeft = sBombersNotebookDayRectRectLeft[BOMBERS_NOTEBOOK_ENTRY_GET_DAY(row, j)] + (startTime / CLOCK_TIME(0, 10));
        entryRectRight = sBombersNotebookDayRectRectLeft[BOMBERS_NOTEBOOK_ENTRY_GET_DAY(row, j)] + (endTime / CLOCK_TIME(0, 10));
```
I think this seems reasonable? There's a few `/ 455` that this could apply to